### PR TITLE
mgr/dashboard: fix dashboard python 3 support

### PR DESCRIPTION
**Pull request by rjfd**
This PR fixes the problems found while running the dashboard in a python 3 only environment.

Signed-off-by: Ricardo Dias <rdias@suse.com>

### CODE REVIEW

**Comment by rjfd** [comment]
PR #21006 is also fixing issues found while testing dashboard in python 3. Therefore, dashboard should be working in python 3 after this and #21006 is merged.

File: qa/tasks/mgr/dashboard/helper.py [hunk 1]
```diff
@@ -78,7 +78,11 @@ def _request(self, url, method, data=None):
         log.info("request %s to %s", method, url)
         if method == 'GET':
             self._resp = self._session.get(url)
-            return self._resp.json()
+            try:
+                return self._resp.json()
+            except ValueError as ex:
+                log.error("response: %s", self._resp.text)
+                log.exception("Failed to decode response: %s", str(ex))
```
Comment (sebastian-philipp) [inline qa/tasks/mgr/dashboard/helper.py]: You don't need to add `str(ex)` to log.exception. it does that automatically. I'd use 
```python
log.exception("Failed to decode response: %s", self._resp.text)
```
instead.

**Also**, you're silently returning `None` here, Can we re-throw the exception here?

File: src/pybind/mgr/dashboard/controllers/rbd.py [hunk 2]
```diff
@@ -76,6 +76,7 @@ def _rbd_list(self, pool_name):
             features = i.features()
             stat['features'] = features
             stat['features_name'] = self._format_bitmask(features)
+            stat['parent_name'] = stat['parent_name'].decode('utf8')
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard/controllers/rbd.py]: Interesting. this seems like a problem of the `rbd` python binding to me.

Comment (rjfd) [reply src/pybind/mgr/dashboard/controllers/rbd.py]: good point! will check rbd python binding implementation

File: qa/tasks/mgr/dashboard/helper.py (hunk 1 above)
Comment (rjfd) [reply qa/tasks/mgr/dashboard/helper.py]: fixed as you suggested

File: src/pybind/mgr/dashboard/controllers/rbd.py (hunk 2 above)
Comment (rjfd) [reply src/pybind/mgr/dashboard/controllers/rbd.py]: I checked the rbd python binding implementation and the `parent_name` and `parent_pool` are already deprecated, so I changed the code to remove those two keys from the dictionary.

**Comment by rjfd** [comment]
QA tests run successfully: http://pulpito.ceph.com/rdias-2018-03-22_16:43:51-rados:mgr-wip-rdias-testing-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
while testing this, I ran into #21021

**Pull request merged**